synapse/lib/httpapi.py

```diff
@@ -149,45 +151,132 @@ async def reqAuthUser(self):
+
+        Args:
+            gen (bool): If set to True, generate a new session if there is no sess cookie.
+
+        Notes:
+            This stores the identifier in the ``sess`` cookie for with a 14 day expiration, stored
+            in the Cell.
+            Valid requests with that ``sess`` cookie will resolve to the same Session object.
+
+        Returns:
+            Sess: A heavy session object. If the sess cookie is invalid or gen is false, this returns None.
+        '''
 
         if self._web_sess is None:
 
             iden = self.get_secure_cookie('sess', max_age_days=14)
 
-            if iden is None and not gen:
-                return None
-
             if iden is None:
-                iden = s_common.guid().encode()
-                opts = {'expires_days': 14, 'secure': True, 'httponly': True}
-                self.set_secure_cookie('sess', iden, **opts)
+                if gen:
+                    iden = s_common.guid().encode()
+                    opts = {'expires_days': 14, 'secure': True, 'httponly': True}
+                    self.set_secure_cookie('sess', iden, **opts)
+                else:
+                    return None
 
             self._web_sess = await self.cell.genHttpSess(iden)
 
         return self._web_sess
 
-    async def user(self):
+    async def user(self):  # pragma: no cover
+        '''
+        Get the heavy User object from the Cell for the authenticated user.
+
+        Deprecated, use the useriden() function instead.
+        '''
+        s_common.deprecated('HandlerBase.user() - use useriden()', eolv='v2.130.0')  # set curv?
+        iden = await self.useriden()
+        if iden is None:
+            return None
+        user = self.cell.auth.user(iden)
+        self._web_user = user
+        return user
+
+    async def useriden(self):
+        '''
+        Get the user iden of the current session user.
 
-        if self._web_user is not None:
-            return self._web_user
+        Note:
+            This function will pull the iden from the current session, or
+            attempt to resolve the useriden with basic authentication.
+
+        Returns:
+            str: The iden of the current session user.
+        '''
+        if self.web_useriden is not None:
+            return self.web_useriden
 
         sess = await self.sess(gen=False)
         if sess is not None:
-            return sess.user
+            iden = sess.info.get('user')
+            name = sess.info.get('name', '<no username>')
+
+            self.web_useriden = iden
+            self.web_username = name
+
+            return iden
+
+        return await self.handleBasicAuth()
+
+    async def handleBasicAuth(self):
+        '''
+        Handle basic authentication in the handler.
+
+        Notes:
+            Implementors may override this to disable or implement their own basic auth schemes.
+            This is expected to set _web_useriden and _web_username upon successful authentication.
```

Comment:
If these are the same attributes set in initialize() they don't have leading underscores?

Reply:
fixed